src/bundle/Command/MultiprocessComand.php

```diff
@@ -0,0 +1,357 @@
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+abstract class MultiprocessComand extends Command
+{
+    /** @var \Ibexa\Contracts\Core\Repository\PermissionResolver */
+    protected $permissionResolver;
+
+    /** @var \Ibexa\Contracts\Core\Repository\UserService */
+    protected $userService;
+
+    /**
+     * @var bool
+     */
+    protected $hasProgressBar;
+
+    /**
+     * @var \Symfony\Component\Console\Helper\ProgressBar
+     */
+    protected $progressBar;
+
+    /**
+     * @var OutputInterface
+     */
+    protected OutputInterface $output;
+
+    /**
+     * @var bool
+     */
+    private bool $dryRun;
+
+    /**
+     * @var int
+     */
+    private int $maxProcesses;
+
+    /**
+     * @var \Symfony\Component\Process\Process[]
+     */
+    private $processes;
+
+    /**
+     * @var string
+     */
+    private mixed $user;
+
+    /**
+     * @var int
+     */
+    private int $iterationCount;
+
+    /**
+     * @var string
+     */
+    private string $environment;
+
+    public function __construct(
+        string $name = null,
+        PermissionResolver $permissionResolver,
+        UserService $userService,
+    ) {
+        $this->permissionResolver = $permissionResolver;
+        $this->userService = $userService;
+        $this->dryRun = false;
+        $this->processes = [];
+
+        parent::__construct($name);
+    }
+
+    public function configure(): void
+    {
+        $this->addOption(
+            'user',
+            'u',
+            InputOption::VALUE_REQUIRED,
+            'Ibexa DXP username',
+            'admin'
+        )
+            ->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Run the converter without writing anything to the database'
+            )
+            ->addOption(
+                'no-progress',
+                null,
+                InputOption::VALUE_NONE,
+                'Disable the progress bar.'
+            )
+            ->addOption(
+                'processes',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Number of child processes to run in parallel for iterations, if set to "auto" it will set to number of CPU cores -1, set to "1" or "0" to disable [default: "auto"]',
+                1
+            )
+                ->addOption(
+                    'iteration-count',
+                    null,
+                    InputOption::VALUE_OPTIONAL,
+                    'Number of objects to process in a single iteration. Set to avoid using too much memory [default: 10000]',
+                    10000
+                );
```

Comment:
Indentation here is broken

Reply:
Fixed in 8264f2d52fb01eaa02784f31a9e28f02e13c6f3e